tests/integrations/starlette/test_starlette.py

```diff
@@ -1150,28 +1166,29 @@ def test_span_origin(sentry_init, capture_events):
         ([range(400, 403), 500, 501], 501, True),
         ([range(400, 403), 500, 501], 503, False),
         ([], 500, False),
+        ([NonIterableContainer(range(500, 600))], 500, True),
+        ([NonIterableContainer(range(500, 600))], 404, False),
```

Comment:
Why do we need the `NonIterableContainer` cases?

Reply:
See this part of the `NonIterableContainer` docstring (lines 1140-1141):

> Used to test backwards compatibility with our old way of defining failed_request_status_codes, which allowed passing in a list of (possibly non-iterable) containers.

The background for why I added this test case is originally I wanted to implement backwards compatibility by just converting any user-provided `list[HttpStatusCodeRange]` to the new `Set[int]` by iterating over the codes and adding all of them to the set. However, this would have been a breaking change from the old API, since an `HttpStatusCodeRange` is defined as an `int` or `Container[int]`, and a `Container[int]` is not necessarily iterable. The `NonIterableContainer` test cases would have failed against my original implementation; I added them to ensure we don't accidentally break this behavior.